sidecar/src/watchdog.rs

```diff
@@ -46,12 +46,44 @@ impl Watchdog {
         let mem_usage_bytes = Arc::new(AtomicUsize::new(0));
         let handle_mem_usage_bytes = mem_usage_bytes.clone();
 
+        let still_alive = Arc::new(AtomicU32::new(0));
+        let still_alive_thread = still_alive.clone();
+
+        let interval = self.interval.period();
+        std::thread::spawn(move || {
+            let mut repeats = false;
```

Comment:
I can see what the code does and this variable is named after that, but maybe call it `stuck` or `maybe_stuck`?